### DISCUSSION TITLE
Pygments (HtmlFormatter?) strips leading and trailing empty lines

### DISCUSSION THREAD

**Discussion by vanschelven** [pygments/pygments#2675] [Q&A] [answered]
When the input code contains leading and trailing empty lines, these are stripped. I would like to preserve them, but AFAICT I cannot instruct `pygments` to do so.

"Empty lines are lines too"; one obvious reason empty lines are important is to keep line numbers correct.

```
>>> import pygments
>>> from pygments import highlight
>>> from pygments.lexers import PythonLexer
>>> from pygments.formatters import HtmlFormatter
>>> 
>>> code = """
... 
... x = 3 
... 
... 
... 
... 
... y = 5
... 
... 
... 
... """
>>> print(highlight(code, PythonLexer(), HtmlFormatter()))
<div class="highlight"><pre><span></span><span class="n">x</span> <span class="o">=</span> <span class="mi">3</span> 




<span class="n">y</span> <span class="o">=</span> <span class="mi">5</span>
</pre></div>
>>> pygments.__version__
'2.16.1'
```

Similar issue which is reportedly fixed: https://github.com/pygments/pygments/issues/818

**Comment by jeanas** [comment, marked as answer]
You should pass `stripnl=False` to `PythonLexer()` as documented [here](https://pygments.org/docs/lexers/).
